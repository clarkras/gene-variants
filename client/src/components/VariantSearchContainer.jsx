import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { connect } from 'react-redux'
import * as actions from '../actions';
import * as selectors from '../selectors'

import GeneSelectDropdown from './GeneSelectDropdown';
import GroupByDropdown from './GroupByDropdown';
import VariantTable from './VariantTable';

export class VariantSearch extends Component {
  static propTypes = {
    setCurrentGene: PropTypes.func.isRequired,
    selectedGene: PropTypes.string,
    groupBy: PropTypes.string,
    variants: PropTypes.arrayOf(PropTypes.object).isRequired,
    genesOptions: PropTypes.array.isRequired,
    fetchVariants: PropTypes.func.isRequired,
    fetchGenesByPrefix: PropTypes.func.isRequired,
  };

  render() {
    const { selectedGene, variants, groupBy, setCurrentGene, selectGroupBy, fetchVariants, fetchGenesByPrefix, genesOptions } = this.props;

    return (
      <div>
        <div className="dropdown-container">
          <GeneSelectDropdown
            selectedGene={selectedGene}
            setCurrentGene={setCurrentGene}
            fetchVariants={fetchVariants}
            fetchGenesByPrefix={fetchGenesByPrefix}
            genesOptions={genesOptions}
          />
          <GroupByDropdown onChange={selectGroupBy} />
        </div>
        <div className="variants-table-container">
          <VariantTable variants={variants} groupBy={groupBy} />
        </div>
      </div>
    );
  }
}

export default connect(
  state => ({
    selectedGene: selectors.getSelectedGene(state),
    groupBy: selectors.getGroupBy(state),
    variants: selectors.getVariants(state),
    genesOptions: selectors.getGenes(state),
  }),
  dispatch => ({
    setCurrentGene: gene => dispatch(actions.setCurrentGene(gene)),
    selectGroupBy: property => dispatch(actions.selectGroupBy(property)),
    fetchVariants: gene => dispatch(actions.fetchVariants(gene)),
    fetchGenesByPrefix: prefix => dispatch(actions.fetchGenesByPrefix(prefix)),
  })
)(VariantSearch)
