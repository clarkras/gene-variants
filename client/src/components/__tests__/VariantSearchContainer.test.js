import React from 'react';
import renderer from 'react-test-renderer';
import { VariantSearch } from '../VariantSearchContainer';

/*global jest, it, expect*/

jest.mock('react-select', () => 'Select');
jest.mock('react-table', () => 'ReactTable');

it('renders without crashing', () => {
  const props = {
    onChange: () => {},
    setCurrentGene: () => {},
    variants: [],
    genesOptions: [],
    fetchVariants: () => {},
    fetchGenesByPrefix: () => {},
    selectGroupBy: () => {},
  };

  const tree = renderer.create(<VariantSearch {...props} />);
  expect(tree).toMatchSnapshot();
});
