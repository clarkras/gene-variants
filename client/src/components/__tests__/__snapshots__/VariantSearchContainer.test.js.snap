// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div
    className="dropdown-container"
  >
    <div
      className="select-container"
    >
      <h3>
        Gene
      </h3>
      <Select
        inputValue=""
        onChange={[Function]}
        onInputChange={[Function]}
        options={Array []}
        placeholder="Enter gene..."
      />
    </div>
    <div
      className="select-container"
    >
      <h3>
        Group by
      </h3>
      <Select
        defaultValue={
          Object {
            "label": "Alias",
            "value": "Alias",
          }
        }
        onChange={[Function]}
        options={
          Array [
            Object {
              "label": "No grouping",
              "value": "",
            },
            Object {
              "label": "Alias",
              "value": "Alias",
            },
            Object {
              "label": "Protein Change",
              "value": "ProteinChange",
            },
            Object {
              "label": "Nucleotide Change",
              "value": "NucleotideChange",
            },
            Object {
              "label": "Region",
              "value": "Region",
            },
            Object {
              "label": "Reported Classification",
              "value": "ReportedClassification",
            },
          ]
        }
        placeholder="Select group column"
      />
    </div>
  </div>
  <div
    className="variants-table-container"
  >
    <ReactTable
      className="-striped -highlight"
      columns={
        Array [
          Object {
            "Header": "Nucleotide Change",
            "accessor": "NucleotideChange",
          },
          Object {
            "Header": "Protein Change",
            "accessor": "ProteinChange",
          },
          Object {
            "Header": "Alias",
            "accessor": "Alias",
          },
          Object {
            "Header": "Region",
            "accessor": "Region",
          },
          Object {
            "Header": "Reported Classification",
            "accessor": "ReportedClassification",
          },
          Object {
            "Header": "Last Evaluated",
            "accessor": "LastEvaluated",
          },
          Object {
            "Header": "Last Updated",
            "accessor": "LastUpdated",
          },
          Object {
            "Cell": [Function],
            "Header": "More Info",
            "accessor": "Source",
          },
        ]
      }
      data={Array []}
      pivotBy={Array []}
    />
  </div>
</div>
`;
